File: BundledMap/app/src/main/java/myco/com/mymapapp/MapLoading/Utilities.java
package myco.com.mymapapp.MapLoading;

import android.util.Log;

import android.content.Context;

import java.io.BufferedInputStream;
import java.io.BufferedOutputStream;
import java.io.ByteArrayOutputStream;
import java.io.File;
import java.io.FileInputStream;
import java.io.InputStream;
import java.io.OutputStream;

public class Utilities {
    private static final String TAG = "LocusLabsMapPack";

    public static String getFileAsString( String filename ) {
        File file = new File( filename );
        if ( !file.exists() ) {
            return null;
        }

        FileInputStream fis = null;
        BufferedInputStream bis = null;
        ByteArrayOutputStream baos = null;
        try {
            fis = new FileInputStream(file);
            bis = new BufferedInputStream(fis);
            baos = new ByteArrayOutputStream();

            int length;
            byte data[] = new byte[2048];

            while ((length = bis.read(data)) != -1) {
                baos.write(data, 0, length);
            }

            baos.flush();

            return baos.toString();
        }
        catch ( Exception exception ) {
            Log.e(TAG, exception.getMessage(), exception);
            return null;
        }
        finally {
            try { if ( fis != null ) fis.close(); } catch ( Exception exception ) {}
            try { if ( bis != null ) bis.close(); } catch ( Exception exception ) {}
            try { if ( baos != null ) baos.close(); } catch ( Exception exception ) {}
        }
    }

    public static void writeInputStreamToOutputStream( InputStream inputStream, OutputStream outputStream  ) throws Exception {
        BufferedOutputStream destination = null;
        try {
            int length;
            byte data[] = new byte[2048];
            destination = new BufferedOutputStream(outputStream);

            while ((length = inputStream.read(data)) != -1) {
                destination.write(data, 0, length);
            }

            destination.flush();
        }
        finally {
            try { if ( outputStream != null ) outputStream.close(); } catch ( Exception exception ) {}
            try { if ( destination != null ) destination.close(); } catch ( Exception exception ) {}
        }
    }
}

File: locuslabs-android-sdk/examples/RecommendedImplementationKotlin/app/src/main/java/com/example/locuslabs/recommendedimplementation/Utilities.kt
package com.example.locuslabs.recommendedimplementation

import android.util.Log

import android.content.Context

import java.io.BufferedInputStream
import java.io.BufferedOutputStream
import java.io.ByteArrayOutputStream
import java.io.File
import java.io.FileInputStream
import java.io.InputStream
import java.io.OutputStream

object Utilities {
    private val TAG = "LocusLabsMapPack"

    fun getFileAsString(filename: String): String? {
        val file = File(filename)
        if (!file.exists()) {
            return null
        }

        var fis: FileInputStream? = null
        var bis: BufferedInputStream? = null
        var baos: ByteArrayOutputStream? = null
        try {
            fis = FileInputStream(file)
            bis = BufferedInputStream(fis)
            baos = ByteArrayOutputStream()

            val data = ByteArray(2048)
            var length: Int = bis.read(data)

            while (length != -1) {
                baos.write(data, 0, length)
                length = bis.read(data)
            }

            baos.flush()

            return baos.toString()
        } catch (exception: Exception) {
            Log.e(TAG, exception.message, exception)
            return null
        } finally {
            try {
                if (fis != null) fis.close()
            } catch (exception: Exception) {
            }

            try {
                if (bis != null) bis.close()
            } catch (exception: Exception) {
            }

            try {
                if (baos != null) baos.close()
            } catch (exception: Exception) {
            }

        }
    }

    @Throws(Exception::class)
    fun writeInputStreamToOutputStream(inputStream: InputStream, outputStream: OutputStream?) {
        var destination: BufferedOutputStream? = null
        try {
            val data = ByteArray(2048)
            var length: Int = inputStream.read(data)
            destination = BufferedOutputStream(outputStream!!)

            while (length != -1) {
                destination.write(data, 0, length)
                length = inputStream.read(data)
            }

            destination.flush()
        } finally {
            try {
                outputStream?.close()
            } catch (exception: Exception) {
            }

            try {
                if (destination != null) destination.close()
            } catch (exception: Exception) {
            }

        }
    }
}
